napari/utils/transforms/transform_utils.py

```diff
@@ -484,7 +484,7 @@ def is_diagonal(matrix, tol=1e-8):
         True if matrix is diagonal, False otherwise.
     """
     if matrix.ndim != 2 or matrix.shape[0] != matrix.shape[1]:
-        raise ValueError("matrix must be square")
+        raise ValueError(trans._("matrix must be square", deferred=True))
```

Comment:
```suggestion
        raise ValueError(trans._("matrix must be square, not has shape {shape}", deferred=True, shape=matrix.shape))
```
For better debug informations